src/Libraries/Microsoft.Extensions.AI.Evaluation/EvaluationResult.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI.Evaluation;
+
+/// <summary>
+/// A collection of one or more <see cref="EvaluationMetric"/>s that represent the result of an evaluation.
+/// </summary>
+public sealed class EvaluationResult
+{
+    /// <summary>
+    /// Gets or sets a collection of one or more <see cref="EvaluationMetric"/>s that represent the result of an
+    /// evaluation.
+    /// </summary>
+#pragma warning disable CA2227
+    // CA2227: Collection properties should be read only.
+    // We disable this warning because we want this type to be fully mutable for serialization purposes and for general
+    // convenience.
+    public IDictionary<string, EvaluationMetric> Metrics { get; set; }
+#pragma warning restore CA2227
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="EvaluationResult"/> class.
+    /// </summary>
+    /// <param name="metrics">
+    /// <para>
+    /// A dictionary containing one or more <see cref="EvaluationMetric"/>s that represent the result of an evaluation.
+    /// </para>
+    /// <para>
+    /// The dictionary is keyed on the <see cref="EvaluationMetric.Name"/>s of the contained
+    /// <see cref="EvaluationMetric"/>s.
+    /// </para>
+    /// </param>
+    [JsonConstructor]
+    public EvaluationResult(IDictionary<string, EvaluationMetric> metrics)
+    {
+        Metrics = metrics;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="EvaluationResult"/> class.
+    /// </summary>
+    /// <param name="metrics">
+    /// An enumeration of <see cref="EvaluationMetric"/>s that represent the result of an evaluation.
+    /// </param>
+    public EvaluationResult(IEnumerable<EvaluationMetric> metrics)
+    {
+        _ = Throw.IfNull(metrics, nameof(metrics));
+
+        var metricsDictionary = new Dictionary<string, EvaluationMetric>();
+
+        foreach (EvaluationMetric metric in metrics)
+        {
+            if (metricsDictionary.ContainsKey(metric.Name))
+            {
+                Throw.ArgumentException(nameof(metrics), $"Cannot add multiple metrics with name '{metric.Name}'.");
+            }
+
+            metricsDictionary[metric.Name] = metric;
```

Comment:
This can be TryAdd on .NET Core. Not sure it's worth ifdef'ing, though.

Reply:
Fixed